### PR TITLE
Add tests for AR node

### PR DESCRIPTION
**Pull request by albertpod**
This pull request includes a comprehensive test set for AR, although it is not fully complete (~80% finished).

### CODE REVIEW

File: test/rules/autoregressive/test_marginals.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                    m_y = NormalMeanPrecision(0.0, 1.0),
+                    m_x = NormalMeanPrecision(0.0, 1.0),
+                    q_θ = NormalMeanPrecision(1.0, 1.0),
+                    q_γ = GammaShapeRate(1.0, 1.0),
+                    meta = ARMeta(Univariate, 1, ARsafe())
+                ),
+                output = MvNormalWeightedMeanPrecision(zeros(2), [2.0 -1.0; -1.0 3.0])
+            )
+        ]
+        # Test for multivariate m_x and m_y are absent due to the numerical instabilities induced by jitters (see marginal rule for autoregressive node)
```
Comment (bvdmitri) [inline test/rules/autoregressive/test_marginals.jl:25]: Do you want to discuss?

File: test/runtests.jl [hunk 2]
```diff
@@ -398,5 +398,11 @@ end
     addtests(testrunner, "rules/transition/test_a.jl")
     addtests(testrunner, "rules/transition/test_in.jl")
 
+    addtests(testrunner, "rules/autoregressive/test_y.jl")
```
Comment (bvdmitri) [inline test/runtests.jl:401]: Please add `nodes/test_autoregressive.jl` as well.

**Review by bvdmitri — CHANGES_REQUESTED** [review]

File: test/runtests.jl (hunk 2 above)
Comment (bvdmitri) [reply test/runtests.jl:401]: My bad, it has been added before.

**Review by bvdmitri — APPROVED** [review]

File: test/rules/autoregressive/test_marginals.jl (hunk 1 above)
Comment (albertpod) [reply test/rules/autoregressive/test_marginals.jl:25]: These huge values correspond to copied states.

**Comment by albertpod** [comment]
ping @bvdmitri

**Pull request merged**